src/test/perl/modules/filetools.pm

```diff
@@ -0,0 +1,31 @@
+package filetools;
+
+use strict;
+use warnings;
+
+use parent qw(Exporter);
+use File::Basename qw(dirname);
+use File::Path qw(mkpath);
+
+our @EXPORT_OK = qw(readfile makefile);
+
+# cannot use mocked filewriter
+sub makefile
+{
+    my $fn = shift;
+    my $dir = dirname($fn);
+    mkpath $dir if ! -d $dir;
+    open(FH, ">$fn");
+    print FH (shift || "ok");
+    close(FH);
```

Comment:
Neither open() nor close() do any error handling which is generally bad, although given this is a unit test helper function, perhaps it's not bad in this specific case?